projects/plugins/jetpack/extensions/blocks/recurring-payments/button/util.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import formatCurrency from '@automattic/format-currency';
+
+/**
+ * Formats a price with the right format for a numeric input value.
+ *
+ * @param {number} price - Price to format.
+ * @param {string} currency - Currency code.
+ * @returns {string} - Formatted price.
+ */
+export const formatPriceForNumberInputValue = ( price, currency ) => {
+	// By using `formatCurrency` we ensure the resulting price contains the relevant decimals for the given currency (i.e. 0.5 > 0.50).
+	return formatCurrency( price, currency, {
+		decimal: '.', // Values for numeric inputs need to use a dot notation for decimals.
+		symbol: '', // Values for numeric inputs cannot contain any currency symbol, only numbers.
+	} );
+};
+
+export const formatProductAmount = product => {
```

Comment:
Maybe we could also add jsdoc to this function so that all the functions defined in this file have them.